pdf/pdflinkarea.cpp

```diff
@@ -148,8 +192,50 @@ void PDFLinkArea::mouseReleaseEvent(QMouseEvent *event)
     if (!rect.contains(event->pos()))
         return;
 
+    // Click action logic in order.
+    // - click on selection;
+    // - unselect if selection is set;
+    // - click on annotation;
+    // - click on link;
+    // - click.
+
     if (!d->canvas) {
         emit clicked(d->clickLocation);
+    } else if (d->clickOnSelection) {
+        emit selectionClicked();
+    } else if (d->selection && d->selection->count() > 0) {
+        d->selection->unselect();
+    } else if (d->annotation != nullptr) {
+        // proxy will be child of this object to avoid memory leak.
```

Comment:
Means also that annotation proxies will be accumulated as long as the document is open, right? Wonder would this work better if ownership is passed to qml engine (QQmlEngine::setObjectOwnership() etc). Didn't test though.